src/vmm/src/config/mod.rs

```diff
@@ -211,6 +217,54 @@ impl TryFrom<&str> for NetConfig {
     }
 }
 
+/// Serial communication configuration
+#[derive(Clone, Debug, PartialEq)]
+pub struct SerialConfig {
+    /// Optional path to pipe the input of the guest OS.
+    pub serial_input: Option<PathBuf>,
+    /// Optional path to pipe the output of the guest OS.
+    pub serial_output: Option<PathBuf>,
+}
+
+impl Default for SerialConfig {
+    fn default() -> Self {
+        SerialConfig {
+            serial_input: None,
+            serial_output: None,
+        }
+    }
+}
+
+impl TryFrom<&str> for SerialConfig {
+    type Error = ConversionError;
+
+    fn try_from(serial_console_str: &str) -> Result<Self, Self::Error> {
+        let mut arg_parser = CfgArgParser::new(serial_console_str);
+
+        let serial_input: Option<String> = arg_parser
+            .value_of("serial_input")
+            .map_err(ConversionError::new_serial)?;
```

Comment:
We can chain an `and_then` here to simplify the code that initializes the `SerialConfig`. If we initialize this directly to an `Option<PathBuf>`, then we will not need the match for building the `SerialConfig`.

We can have it as:
```rust
        let serial_input = arg_parser
            .value_of("serial_input")
            .map_err(ConversionError::new_serial)?
            .and_then(|inner: String| Some(PathBuf::from(inner)));
```